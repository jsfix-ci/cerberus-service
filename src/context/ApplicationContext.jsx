import React, { createContext, useRef } from 'react';

const ApplicationContext = createContext({});

/**
 * This context file contains is for holding state data that does not change
 * frequently and can be accessible throughout the app.
 */
const ApplicationContextProvider = ({ children }) => {
  const _refDataAirlineCodes = useRef([]);
  const _tisCache = useRef({});

  const refDataAirlineCodes = () => {
    return _refDataAirlineCodes.current;
  };

  const setRefDataAirlineCodes = (value) => {
    _refDataAirlineCodes.current = value;
  };

  const tisCache = () => {
    return _tisCache.current;
  };

  const setTisCache = (value) => {
    _tisCache.current = value;
  };

  return (
    <ApplicationContext.Provider value={{ refDataAirlineCodes,
      setRefDataAirlineCodes,
      tisCache,
      setTisCache }}
    >
      {children}
    </ApplicationContext.Provider>
  );
};

export { ApplicationContext, ApplicationContextProvider };
