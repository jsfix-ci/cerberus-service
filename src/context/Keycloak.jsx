import React, {
  createContext, useState, useContext, useEffect,
} from 'react';
import Keycloak from 'keycloak-js';
import { useInterval } from 'react-use';

import config from '../utils/config';

const KeycloakContext = createContext();

const KeycloakProvider = ({ children }) => {
  const [keycloak, setKeycloak] = useState(null);
  const [refreshToken, setRefreshToken] = useState(false);
  const keycloakInstance = Keycloak(config.keycloak.clientConfig);

  useInterval(() => {
    /**
    * isTokenExpired(minValidity)
    * Returns true if the token has less than minValidity seconds left before
    * it expires (minValidity is optional, if not specified 0 is used)
    */
    if (keycloak.isTokenExpired(config.keycloak.minExpiryValidity)) {
      setRefreshToken(!refreshToken);
    }
  }, config.keycloak.pollingInterval);

  useEffect(() => {
    if (keycloak) {
      keycloak
        .updateToken()
        .catch(() => {
          keycloak.logout();
        });
    }
  }, [refreshToken]);

  useEffect(() => {
    keycloakInstance.init(config.keycloak.initOptions).then((authenticated) => {
      if (authenticated) {
        setKeycloak(keycloakInstance);
      } else {
        keycloakInstance.login();
      }
    });
  }, []);

  return <KeycloakContext.Provider value={keycloak}>{children}</KeycloakContext.Provider>;
};

const useKeycloak = () => useContext(KeycloakContext);

export { KeycloakContext, KeycloakProvider, useKeycloak };
