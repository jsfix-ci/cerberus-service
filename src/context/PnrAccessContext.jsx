import React, { useState, createContext } from 'react';

import { LOCAL_STORAGE_KEYS } from '../utils/constants';

const PnrAccessContext = createContext({});

const PnrAccessProvider = ({ children }) => {
  const storedUserSession = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEYS.PNR_USER_SESSION_ID));
  const [canViewPnrData, setCanViewPnrData] = useState(storedUserSession?.requested);

  const setViewPnrData = (viewData) => setCanViewPnrData(viewData);

  return (
    <PnrAccessContext.Provider value={{ canViewPnrData, setViewPnrData }}>
      {children}
    </PnrAccessContext.Provider>
  );
};

export { PnrAccessContext, PnrAccessProvider };
