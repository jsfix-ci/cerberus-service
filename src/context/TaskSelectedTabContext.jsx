import React, { useState, createContext } from 'react';

const TaskSelectedTabContext = createContext({});

const TaskSelectedTabProvider = ({ children }) => {
  const [taskManagementTabIndex, setTaskManagementTabIndex] = useState(0);

  const selectTaskManagementTabIndex = (tabIndex) => setTaskManagementTabIndex(tabIndex);

  return (
    <TaskSelectedTabContext.Provider
      value={{ taskManagementTabIndex, selectTaskManagementTabIndex }}
    >
      {children}
    </TaskSelectedTabContext.Provider>
  );
};

export { TaskSelectedTabContext, TaskSelectedTabProvider };
