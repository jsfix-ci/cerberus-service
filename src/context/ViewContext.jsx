import React, { createContext, useEffect, useRef } from 'react';
import { useLocation } from 'react-router-dom';
import { PATHS, VIEW } from '../utils/constants';

const ViewContext = createContext({});

/**
 * This context file contains is for holding state data that does not change
 * frequently and can be accessible throughout the app.
 */
const ViewProvider = ({ children }) => {
  const _view = useRef('');
  const location = useLocation();

  const getView = () => {
    return _view.current;
  };

  const setView = (value) => {
    _view.current = value;
  };

  useEffect(() => {
    const determineViewByLocation = (_location) => {
      if (PATHS.RORO_PATHS.includes(_location)) {
        setView(VIEW.RORO);
      }
      if (PATHS.AIRPAX_PATHS.includes(_location)) {
        setView(VIEW.AIRPAX);
      }
      if (!PATHS.AIRPAX_PATHS.includes(_location) && !PATHS.RORO_PATHS.includes((_location))) {
        setView(VIEW.NONE);
      }
    };

    determineViewByLocation(location?.pathname);
  }, [location?.pathname]);

  return (
    <ViewContext.Provider value={{ getView }}>
      {children}
    </ViewContext.Provider>
  );
};

export default ViewProvider;

export { ViewContext };
