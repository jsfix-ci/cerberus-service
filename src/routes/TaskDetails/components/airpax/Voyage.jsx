import React, { useContext } from 'react';
import { DATE_FORMATS } from '../../../../utils/constants';
import { ApplicationContext } from '../../../../context/ApplicationContext';

import { JourneyUtil, MovementUtil } from '../../../../utils';
import renderBlock from '../../helper/common';

const Voyage = ({ version }) => {
  const journey = JourneyUtil.get(version);
  const flight = MovementUtil.movementFlight(version);
  const { refDataAirlineCodes } = useContext(ApplicationContext);
  return (
    <div className="task-details-container bottom-border-thin govuk-!-margin-bottom-2">
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Voyage</h3>
      <div className="govuk-task-details-grid-column">
        {renderBlock('Departure from', [JourneyUtil.departureLoc(journey), MovementUtil.formatLoc(JourneyUtil.departureLoc(journey))])}
        {renderBlock('Departure time', [JourneyUtil.formatDepartureTime(journey, DATE_FORMATS.LONG_DAY_DATE)])}
        {renderBlock('Arrival at', [JourneyUtil.arrivalLoc(journey), MovementUtil.formatLoc(JourneyUtil.arrivalLoc(journey))])}
        {renderBlock('Arrival time', [JourneyUtil.formatArrivalTime(journey, DATE_FORMATS.LONG_DAY_DATE)])}
        {renderBlock('Scheduled flight time', [JourneyUtil.formatFlightTime(journey)])}
        {renderBlock('Flight code', [MovementUtil.flightNumber(flight)])}
        {renderBlock('Operator', [MovementUtil.airlineName(MovementUtil.airlineOperator(flight), refDataAirlineCodes())])}
      </div>
    </div>
  );
};

export default Voyage;
