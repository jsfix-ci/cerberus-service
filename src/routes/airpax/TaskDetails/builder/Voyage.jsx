import React, { useContext } from 'react';
import { LONG_DAY_DATE_FORMAT } from '../../../../constants';
import { ApplicationContext } from '../../../../context/ApplicationContext';

import { MovementUtil } from '../../utils';
import renderBlock from './helper/common';

const Voyage = ({ version }) => {
  const journey = MovementUtil.movementJourney(version);
  const flight = MovementUtil.movementFlight(version);
  const { refDataAirlineCodes } = useContext(ApplicationContext);
  return (
    <div className="task-details-container bottom-border-thin govuk-!-margin-bottom-2">
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Voyage</h3>
      <div className="govuk-task-details-grid-column">
        {renderBlock('Departure from', [MovementUtil.departureLoc(journey), MovementUtil.formatLoc(MovementUtil.departureLoc(journey))])}
        {renderBlock('Departure time', [MovementUtil.formatDepartureTime(journey, LONG_DAY_DATE_FORMAT)])}
        {renderBlock('Arrival at', [MovementUtil.arrivalLoc(journey), MovementUtil.formatLoc(MovementUtil.arrivalLoc(journey))])}
        {renderBlock('Arrival time', [MovementUtil.formatArrivalTime(journey, LONG_DAY_DATE_FORMAT)])}
        {renderBlock('Scheduled flight time', [MovementUtil.formatFlightTime(journey)])}
        {renderBlock('Flight code', [MovementUtil.flightNumber(flight)])}
        {renderBlock('Operator', [MovementUtil.airlineName(MovementUtil.airlineOperator(flight), refDataAirlineCodes())])}
      </div>
    </div>
  );
};

export default Voyage;
