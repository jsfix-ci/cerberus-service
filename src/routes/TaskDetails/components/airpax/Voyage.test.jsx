import React from 'react';
import renderer from 'react-test-renderer';

import Voyage from './Voyage';
import { ApplicationContext } from '../../../../context/ApplicationContext';

import { versions } from '../../../../__fixtures__/airpax-task-details-versions.fixture.json';
import refDataAirlineCodes from '../../../../__fixtures__/airpax-airline-codes.json';

describe('Voyage', () => {
  it('should render the component without crashing', () => {
    const tree = renderer.create(
      <ApplicationContext.Provider value={{ refDataAirlineCodes: jest.fn().mockReturnValue(refDataAirlineCodes) }}>
        <Voyage version={versions[0]} />
      </ApplicationContext.Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
