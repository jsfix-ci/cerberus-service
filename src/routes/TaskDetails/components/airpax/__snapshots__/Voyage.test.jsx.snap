// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voyage should render the component without crashing 1`] = `
<div
  className="task-details-container bottom-border-thin govuk-!-margin-bottom-2"
>
  <h3
    className="govuk-heading-m govuk-!-margin-top-0"
  >
    Voyage
  </h3>
  <div
    className="govuk-task-details-grid-column"
  >
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Departure from
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        YYC
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Calgary, Canada
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Departure time
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Wed 3 Oct 2018 at 18:32
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Arrival at
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        LHR
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        London, United Kingdom
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Arrival time
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Wed 3 Oct 2018 at 21:19
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Scheduled flight time
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        2h 46m
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Flight code
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        AC0850
      </p>
    </div>
    <div
      className="govuk-!-margin-bottom-2"
    >
      <p
        className="govuk-!-margin-bottom-0 font__light"
      >
        Operator
      </p>
      <p
        className="govuk-!-margin-bottom-0 font__bold"
      >
        Unknown
      </p>
    </div>
  </div>
</div>
`;
